src/libos/src/net/socket/unix/stream/endpoint.rs

```diff
@@ -108,3 +141,51 @@ impl Inner {
 // TODO: Add SO_SNDBUF and SO_RCVBUF to set/getsockopt to dynamcally change the size.
 // This value is got from /proc/sys/net/core/rmem_max and wmem_max that are same on linux.
 pub const DEFAULT_BUF_SIZE: usize = 208 * 1024;
+
+/// An observer used to observe both reader and writer of the endpoint. It also contains a
+/// notifier that relays the notification of the endpoint.
+pub(super) struct RelayNotifier {
+    notifier: IoNotifier,
+    endpoint: SgxMutex<Option<Endpoint>>,
+}
```

Comment:
There is a memory leakage because `Endpoint` and `RelayNotifier` have references to each other.

Reply:
The relationship is below:
stream(relaynotifier), stream(endpoint), relaynotifer->endpoint
Will endpoint reference relaynotifier?

Reply:
Great. Then we don't have a problem here.